worker/backend/backend.go

```diff
@@ -112,36 +138,41 @@ func (b *Backend) Create(gdnSpec garden.ContainerSpec) (container garden.Contain
 // TODO: list the resources that can be acquired during the lifetime of a container.
 //
 // Errors:
-// * TODO.
+// * Container handle does not exist
+// * Container task was not successfully spun down prior to container deletion.
+//   Reasons include task not found, termination signal failed, or task deletion failed.
+// * Destroy request to client failed
+//
 func (b *Backend) Destroy(handle string) error {
-	ctx := namespaces.WithNamespace(context.Background(), b.namespace)
-	const maxTaskKillWaitTime = 10 * time.Second
-
 	if handle == "" {
-		return InputValidationError{}
+		return InputValidationError{Message: "handle is empty"}
 	}
 
-	container, err := b.client.GetContainer(ctx, handle)
+	ctx := namespaces.WithNamespace(context.Background(), b.namespace)
+	ctxWithTimeout, _ := context.WithTimeout(ctx, b.clientTimeout)
+
+	container, err := b.client.GetContainer(ctxWithTimeout, handle)
 	if err != nil {
-		return err
+		return ClientError{ InnerError: err }
 	}
 
-	task, err := container.Task(ctx, nil)
+	task, err := container.Task(ctxWithTimeout, nil)
 	if err != nil {
 		if !errdefs.IsNotFound(err) {
-			return err
+			return ClientError { InnerError: err }
+		}
+	} else {
+		err = killTasks(ctxWithTimeout, task)
+		if err != nil {
+			return ClientError{ InnerError: err }
 		}
-
-		return b.client.Destroy(ctx, handle)
 	}
 
-	timeDelimitedContext, _ := context.WithTimeout(ctx, maxTaskKillWaitTime)
-	err = killTasks(timeDelimitedContext, task)
+	err = b.client.Destroy(ctxWithTimeout, handle)
```

Comment:
is the reason we switched from killTasks to Destroy to avoid killing gracefully?

Reply:
It's just a refactor - we only want to call `killTasks` if we can reference the container without error (line 165). This is kind of a complicated piece of logic that we attempted to explain better in the unit tests -- we moved `killTasks` up, so that Destroy gets in both of these circumstances:

1. A container is found but no Task was ever started inside it (line 161). This means that container creation was somehow interrupted, maybe by a network partition, so we want this container to be cleanly deletable later on by a GC. We fence off the "task does not exist" genre of error in 161.
1. A container is found and its Task was successfully stopped.

In both of these cases we want Destroy to be the next thing that happens. If no errors trigger early returns, then this should be the control flow now!